segmentstore/server/host/src/main/java/io/pravega/service/server/host/stat/AutoScaleProcessor.java

```diff
@@ -185,18 +186,29 @@ private void triggerScaleDown(String streamSegmentName, boolean silent) {
 
     private CompletableFuture<Void> writeRequest(ScaleEvent event) {
         CompletableFuture<Void> result = new CompletableFuture<>();
+        CompletableFuture<Void> writeComplete = new CompletableFuture<>();
         try {
-            CompletableFuture.runAsync(() -> {
+            AckFuture ackFuture = writer.get().writeEvent(event.getKey(), event);
+            ackFuture.addListener(() -> writeComplete.complete(null), executor);
+            writeComplete.thenAcceptAsync((Void v) -> {
                 try {
-                    writer.get().writeEvent(event.getKey(), event).get();
-                    result.complete(null);
-                } catch (InterruptedException | ExecutionException e) {
-                    log.error("error sending request to requeststream {}", e);
+                    ackFuture.get();
+                } catch (ExecutionException | InterruptedException e) {
```

Comment:
there is also execution exception. 
I have changed it to FutureHelpers.getAndHandleException.